compiler/rustc_ast/src/token.rs

```diff
@@ -590,10 +590,7 @@ impl Token {
 
     /// Returns `true` if the token is a non-raw identifier for which `pred` holds.
     pub fn is_non_raw_ident_where(&self, pred: impl FnOnce(Ident) -> bool) -> bool {
-        match self.ident() {
-            Some((id, false)) => pred(id),
-            _ => false,
-        }
+        matches!(self.ident(), Some((id, false)) if pred(id))
```

Comment:
idk, using guards in `matches!` seems not great for readability to me. I would just keep the current code.

Reply:
Dropped.